torch/_inductor/kernel/mm.py

```diff
@@ -89,7 +89,7 @@
             b = tl.load(B, mask=rk[:, None] < k, other=0.)
         if B_PROLOGUE_CAST_TYPE is not None:
             b = b.to(B_PROLOGUE_CAST_TYPE)
-        acc += tl.dot(a, b, allow_tf32=ALLOW_TF32)
+        acc += tl.dot(a, b, input_precision=INPUT_PRECISION)
```

Comment:
We probably need to be backward compatible with triton versions that don't have `input_precision` yet.

Reply:
Do we? I have not seen a policy regarding this, but our published wheels metadata have an exact version pin to the `pytorch-triton` version built from the commit pin at release. 

I think we only need to support the pinned version of triton.

Reply:
There are several triton pins that aren't all kept in sync. e.g. meta-internal pin and triton-rocm pin

Reply:
For internal pins, both the trunk and the Rocm fork is being upgraded to support `input_precision`